src/libs/PaymentUtils.js

```diff
@@ -71,6 +92,42 @@ function formatPaymentMethods(bankAccountList, cardList, payPalMeData = null) {
     return combinedPaymentMethods;
 }
 
+/**
+ * Gets the selected plaid bank account object from plaidData
+ * @param {Object} personalBankAccount
+ * @param {Object} plaidData
+ * @returns {Object}
+ */
+function getPendingBankAccount(personalBankAccount = {}, plaidData = {}) {
+    // Get selected bank account ID if present
+    const plaidAccountID = lodashGet(personalBankAccount, 'selectedPlaidAccountID', 0);
+    if (!plaidAccountID) {
+        return {};
+    }
+
+    // Get bank accounts returned by Plaid
+    const plaidBankAccounts = lodashGet(plaidData, 'bankAccounts', []);
+    if (_.isEmpty(plaidBankAccounts)) {
+        return {};
+    }
+
+    // Get that bank account object by ID
+    const pendingBankAccount = _.findWhere(plaidBankAccounts, {plaidAccountID}) || {};
+    if (_.isEmpty(pendingBankAccount)) {
+        return {};
+    }
+
+    // Get errors
```

Comment:
Why are we reformatting the errors here?

Reply:
This is how we use to. How do we pass back errors now? Atm we don't seem to pass back any errors within the item itself. 

We pass errors back in timestamp => errorMessage format, but the item object takes an array of error messages

Reply:
Sorry, I don't think I'm following the logic in this function. So we're basically fetching the `personalBankAccount` currently being setup, taking the errors and adding it to `pendingBankAccount` (retrieved from the Plaid data). And we're doing this because we need some additional properties like the account number, address name, etc. in `formatPaymentMethods`. Is that right?

Perhaps an alternative here would be to take the properties we need from `pendingBankAccount` and add them to `personalBankAccount` instead. So we won't have to do any reformatting of the errors. Not a major win but maybe would make this more understandable?

Reply:
Ah, so we have bank account info in `plaidData`. Stuff like unmasked account numbers and such. Data we need to pass to PHP when trying to add a bank account. 

_Currently_, when you try to add a bank account, we let the data sit in `plaidData` and just save `selectePlaidAccountID` in the onyxkey `personalBankAccount`. We currently use that to find the bank account they selected in `plaidData`. And we then format that data so that it would appear in the payment list.

You are suggesting that upon submission, we save data we need, formatted already, into `personalBankAccount`. So when we show the payment methods, we don't have to do anything here. 

I'm good with that. My argument against that is, we would be doing some sort of "massaging" somewhere in the client side, because its client side data. So instead of doing all the formatting and such here, we'd do it in `BankAccounts.addPersonalBankAccount`. It may make this method cleaner, but still somewhat convoluted? And I think it'll make it harder to follow.

Reply:
Thanks for clarifying! I like your suggestion:

```
So instead of doing all the formatting and such here, we'd do it in `BankAccounts.addPersonalBankAccount`
```

I think perhaps the flow would be easier to understand if we did the above. Would something like this work:
1. The user submits the add bank account form
2. We fetch the all we need from `plaidData` and store it in the `personalBankAccount` key
3. We can display the pending bank account in the payment methods list without needing any more formatting in `formatPaymentMethods(...)`